### PR TITLE
Add Simple Integration test (CLI - 'otx train')

### PR DESCRIPTION
**Pull request by harimkang**
### Summary

- Add CLI Integration Test using recipes
    - 1 epochs training with all recipes
    - Currently, only classification and detection work
- Add Classification test assets dataset: `tests/assets/classification_dataset`
- Add `trainer.device=auto` setting
- Add tox integration-test
- Add integration test into CI
- Fix install issue of case without 'version.json' in '/cuda/'

### How to test

- TOX test
`tox -vv -e integration-test`

- Local Test
`python -m pytest tests/integration/cli/test_cli.py`
image

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by jaegukhyun — DISMISSED** [review]

File: tests/integration/cli/test_cli.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# This assumes have OTX installed in environment.
+otx_module = importlib.import_module("otx")
+RECIPE_PATH = Path(inspect.getfile(otx_module)).parent / "recipe"
+RECIPE_LIST = [str(_.relative_to(RECIPE_PATH)) for _ in RECIPE_PATH.glob("**/*.yaml")]
+
+# [TODO]: This is a temporary approach.
+DATASET = {
+    "classification": {
+        "data_dir": "tests/assets/classification_dataset",
+        "overrides": ["model.otx_model.config.head.num_classes=2"],
```
Comment (sungmanc) [inline tests/integration/cli/test_cli.py]:
```suggestion
        "overrides": ["model.otx_model.config.head.num_classes=2", "model.otx_model.config.data_preprocessor.num_classes=2"],
```

Comment (harimkang) [reply tests/integration/cli/test_cli.py]: Thanks! - https://github.com/openvinotoolkit/training_extensions/pull/2695/commits/b5399f75f1f726cbae28339c8a83e455864acb02

**Review by sungmanc — APPROVED** [review]
LGTM

**Review by jaegukhyun — APPROVED** [review]

**Review by kprokofi — APPROVED** [review]
In general, LGTM. But it might be worth extending the tests by 2 epochs instead of 1 to make sure that any processes between epochs also work.
What do you think?

**Comment by harimkang** [comment]
> In general, LGTM. But it might be worth extending the tests by 2 epochs instead of 1 to make sure that any processes between epochs also work. What do you think?

I think it's a good idea, but for now I'll merge this PR for another task model recipe integration test and create a new PR after we discuss it together.

File: tests/integration/cli/test_helper.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    async def run_and_capture(cmd: list) -> tuple:
+        environ = os.environ.copy()
+        environ["PYTHONUNBUFFERED"] = "1"
+        process = await asyncio.create_subprocess_exec(
+            *cmd,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+            env=environ,
+            **kwargs,
+        )
```
Comment (vinnamkim) [inline tests/integration/cli/test_helper.py:55]: I do not want to spawn a separate process for the integration test. This is because it makes a developer harder to debug if a problem is found during the integration test. Please replace it with executing the CLI Python function in the main process.

File: tests/integration/cli/test_cli.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    task = recipe.split("/")[0]
+    model_name = recipe.split("/")[1].split(".")[0]
+    tmp_path = tmp_path / f"otx_train_{model_name}"
+    command_line = [
+        "otx", "train",
+        f"+recipe={recipe}",
+        f"base.data_dir={DATASET[task]['data_dir']}",
+        "trainer.max_epochs=1",
+        "trainer.check_val_every_n_epoch=1",
+        "trainer=auto",
```
Comment (vinnamkim) [inline tests/integration/cli/test_cli.py:55]: Rather than manually overriding at the variable level, consider to override this https://github.com/openvinotoolkit/training_extensions/blob/v2/src/otx/config/debug/default.yaml

File: src/otx/config/trainer/auto.yaml
Comment (vinnamkim) [inline src/otx/config/trainer/auto.yaml:1]: Why is this needed?

File: tests/integration/cli/test_cli.py (hunk 3 above)
Comment (harimkang) [reply tests/integration/cli/test_cli.py:55]: Thanks for the comment. I will create a PR to reflect your comments. I have a question regarding this comment. I want to create a config file called `debug/intg_test.yaml` and want to apply it to test. 
Can you tell me what args should be added to override the config in this debug folder in the cli?

File: src/otx/config/trainer/auto.yaml
Comment (harimkang) [reply src/otx/config/trainer/auto.yaml:1]: I wanted to make sure that CI would work whether the test was using cpu or gpu, so I thought using the `auto` provided by lightning for the test would be appropriate. What do you think? This too can be modified as per your suggestion.

File: tests/integration/cli/test_cli.py (hunk 3 above)
Comment (vinnamkim) [reply tests/integration/cli/test_cli.py:55]: I guess that you need to add `debug` optional field to `TrainConfig` as same as `recipe`.
image
Then, it might be working by adding argument `+debug=intg_test`. This is exactly same with what we do with `+recipe=???`.

File: src/otx/config/trainer/auto.yaml
Comment (vinnamkim) [reply src/otx/config/trainer/auto.yaml:1]: Is that possible by giving `trainer=cpu` or `trainer=gpu` explicitly outside? If you set it as `auto`, then it seems not traceable which device we used for the tests.

File: tests/integration/cli/test_cli.py (hunk 3 above)
Comment (harimkang) [reply tests/integration/cli/test_cli.py:55]: Thank you. I'll give it a try.

File: src/otx/config/trainer/auto.yaml
Comment (harimkang) [reply src/otx/config/trainer/auto.yaml:1]: For test, yes i understand, I'm going to modify it in intg-test to gpu for now. Do you want to remove this file?
I guess `auto` might be a good option for the user as well.

Comment (vinnamkim) [reply src/otx/config/trainer/auto.yaml:1]: Then, let it leave as is. However, the decision for which device will be used should be explicit in the tests. This is what I only concern.

**Comment by harimkang** [comment]
@vinnamkim I create next PR https://github.com/openvinotoolkit/training_extensions/pull/2701
